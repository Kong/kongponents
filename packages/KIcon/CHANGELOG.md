# @kongponents/kicon

## Table of Contents

- [0.2.3](#023)

### 0.2.3
 - Adds check to render when client is available & adds `transform-box` to spinner class [#127](https://github.com/Kong/kongponents/pull/127)
