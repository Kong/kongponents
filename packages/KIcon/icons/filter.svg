<svg width="14" height="11" xmlns="http://www.w3.org/2000/svg"><path d="M8 9v2H6V9h2zm2-3v2H4V6h6zm2-3v2H2V3h10zm2-3v2H0V0h14z" fill="#000" fill-rule="evenodd" fill-opacity=".45"/></svg>