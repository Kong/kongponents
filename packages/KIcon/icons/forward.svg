<svg width="15" height="9" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M.293 4.496L4.09.7v3.147h10.3v1.3H4.09v3.147L.293 4.496z" fill="#000" fill-opacity=".25"/></svg>