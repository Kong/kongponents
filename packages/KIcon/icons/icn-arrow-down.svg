<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 16 17">
    <title>Down</title>
    <path fill="#A3BBCC" fill-rule="evenodd" d="M11.2 11c.5-.6.3-1-.4-1H9.4V4a1.3 1.3 0 1 0-2.6 0v6H5.2c-.7 0-.9.4-.4 1l2.4 2.8a1 1 0 0 0 1.7 0l2.3-2.8Z" clip-rule="evenodd"/>
</svg>