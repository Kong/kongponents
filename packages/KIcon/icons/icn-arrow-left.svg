<svg viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
  <title>Back</title>
  <path d="M12.17 8.75H3M6.75 5 3 8.75l3.75 3.75" fill="none" stroke="#A3BBCC" stroke-width="1.25" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
