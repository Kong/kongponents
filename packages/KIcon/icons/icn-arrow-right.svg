<svg viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="M3 8.75h9.17M8.42 5l3.75 3.75-3.75 3.75" fill="none" stroke="#A3BBCC" stroke-width="1.25" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
