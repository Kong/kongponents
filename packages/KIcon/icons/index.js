import collapseExpand from './icn-collapse-expand.svg'
import gateway from './icn-gateway.svg'
import gear from './icn-gear.svg'
import info from './icn-info.svg'
import portal from './icn-dev-portal.svg'
import security from './icn-security.svg'
import workspaces from './icn-workspaces.svg'
import workspacesCollapsed from './icn-collapse-workspaces.svg'
import vitals from './icn-vitals.svg'
import back from './icn-back.svg'
import search from './icn-search.svg'
import trash from './icn-trash.svg'
import dashboard from './icn-dashboard.svg'
import support from './icn-support.svg'
import featureRequest from './icn-feature-request.svg'
import externalLink from './icn-external-link.svg'
import spinner from './icn-spinner.svg'

export default {
  back,
  collapseExpand,
  dashboard,
  externalLink,
  featureRequest,
  gateway,
  gear,
  info,
  portal,
  search,
  security,
  spinner,
  support,
  trash,
  workspaces,
  workspacesCollapsed,
  vitals
}
