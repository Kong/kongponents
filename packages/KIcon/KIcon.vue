<template>
  <svg
    :height="setSize || height"
    :width="setSize || width"
    :viewBox="viewbox"
    :class="`kong-icon-${icon}`"
    class="kong-icon"
    role="img"
  >
    <title>{{ icon }}</title>
    <g>
      <path
        v-for="(path, idx) in paths"
        :key="path.d"
        v-bind="attributes[idx]"/>
    </g>
  </svg>
</template>

<script>
import icons from './icons'
const iconNames = Object.keys(icons)

export default {
  name: 'KIcon',
  props: {
    /**
     * Checks for valid icon name<br>
     * 'back' | 'collapseExpand' | 'gateway' | 'gear' | 'info' | 'portal' | 'search' | 'security' | 'workspaces' | 'workspacesCollapsed' | 'vitals'
     */
    icon: {
      type: String,
      default: '',
      validator: function (value) {
        return iconNames.indexOf(value) !== -1
      },
      required: true
    },
    /**
     * Optional - Overrides default height and width with equal value
     */
    size: {
      type: String,
      default: '24'
    },
    /**
     * Optional - Sets Fill color
     */
    color: {
      type: String,
      default: null
    },
    /**
     * Optional - Defines viewbox dimensions
     */
    viewBox: {
      type: String,
      default: '0 0 24 24'
    }
  },

  computed: {
    iconSVG () {
      return icons[this.icon]
    },
    doc () {
      return new DOMParser().parseFromString(this.iconSVG, 'image/svg+xml')
    },
    svg () {
      return this.doc.getElementsByTagName('svg')[0]
    },
    paths () {
      return this.doc.querySelectorAll('path').length ? Array.from(this.doc.querySelectorAll('path')) : console.warn('(KIcon) Warning: SVG Path not found')
    },
    attributes () {
      if (this.paths) {
        let attributes = []

        this.paths.forEach(path => {
          let pathAttributes = {}
          let attrs = Array.from(path.attributes)

          attrs.forEach((attr) => {
            const { value, name } = attr

            if (name === 'fill' && this.color) {
              pathAttributes[name] = this.color
            } else {
              pathAttributes[name] = value
            }
          })

          attributes.push(pathAttributes)
        })

        return attributes
      }
    },
    width () {
      return this.svg.getAttribute('width')
    },
    height () {
      return this.svg.getAttribute('height')
    },
    setSize () {
      return this.size || this.svg.getAttribute('width')
    },
    viewbox () {
      return this.viewBox || this.svg.getAttribute('viewBox')
    }
  }
}
</script>

<style lang="scss" scoped>
.kong-icon {
  &.kong-icon-spinner g {
    transform-origin: 50% 50%;
    animation: spin 1.2s infinite linear;
  }
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(1turn); }
}
</style>
