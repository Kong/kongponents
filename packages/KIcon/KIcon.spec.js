import { mount } from '@vue/test-utils'
import KIcon from '@/KIcon/KIcon'
import icons from './icons'
const iconNames = Object.keys(icons)

let rendersIcon = (icon) => {
  it(`renders ${icon} icon`, () => {
    const wrapper = mount(KIcon, {
      propsData: {
        'icon': `${icon}`
      }
    })

    expect(wrapper.find('title').text()).toEqual(expect.stringContaining(`${icon}`))
    expect(wrapper.html()).toMatchSnapshot()
  })
}

describe('KIcon', () => {
  iconNames.map(icon => {
    rendersIcon(icon)
  })

  it('renders icon with red fill', () => {
    const wrapper = mount(KIcon, {
      propsData: {
        'icon': 'portal',
        'color': 'red'
      }
    })

    const path = wrapper.find('path').attributes()

    expect(path.fill).toEqual('red')
  })

  it('renders 32x32 sized icon', () => {
    const wrapper = mount(KIcon, {
      propsData: {
        'icon': 'portal',
        'size': '32'
      }
    })

    expect(wrapper.attributes().width).toEqual('32')
    expect(wrapper.attributes().height).toEqual('32')
  })
})
