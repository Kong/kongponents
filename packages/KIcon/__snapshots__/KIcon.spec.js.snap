// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KIcon renders back icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-back">
  <title>back</title>
  <g>
    <path fill="#000" fill-opacity=".25" fill-rule="evenodd" d="M16 5H5v3L0 4l3.371-2.697L5 0v3h11z"></path>
  </g>
</svg>
`;

exports[`KIcon renders collapseExpand icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-collapseExpand">
  <title>collapseExpand</title>
  <g>
    <path fill="#A3BBCC" fill-rule="evenodd" d="M0 0h2v16H0V0zm16 7v2H9v3L4 8l5-4v3h7z"></path>
  </g>
</svg>
`;

exports[`KIcon renders dashboard icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-dashboard">
  <title>dashboard</title>
  <g>
    <path fill="#A4BBCB" fill-rule="nonzero" d="M13 10l3-3c.6902336-.73713266 1.4429032-1.26058218 2-1 .2605822.55709682-.2628673 1.30976636-1 2l-3 3c.1531248.4707586.1875961.6844547 0 1 .1875961 1.0625747-.7498292 2-2 2-1.0625747 0-2-.9374253-2-2 0-1.2501708.9374253-2.1875961 2-2 .3155453-.1875961.5292414-.1531248 1 0zm8 10c-.08276.0634515-.7131836.1223062-1 0-.5634515-.58276-.6223062-1.2131836 0-2 1.1963422-1.4063656 2-3.6213758 2-6 0-5.5228475-4.4771525-10-10-10S2 6.4771525 2 12c0 2.3848064.80099865 4.5939865 2 6 .62209487.7857069.56386353 1.4161884 0 2-.28570688.1220949-.91618845.0638635-1 0-2.03512593-2.4863139-3-5.1474677-3-8C0 5.372583 5.372583 0 12 0s12 5.372583 12 12c0 2.8461786-.967377 5.5124254-3 8zM9 16h6c.5522847 0 1 .4477153 1 1s-.4477153 1-1 1H9c-.55228475 0-1-.4477153-1-1s.44771525-1 1-1z"></path>
  </g>
</svg>
`;

exports[`KIcon renders externalLink icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-externalLink">
  <title>externalLink</title>
  <g>
    <path fill="#000" fill-opacity=".25" fill-rule="evenodd" d="M8.24308 1.5C7.83269 1.5 7.5 1.16710138 7.5.75c0-.41421356.33823-.75.74308-.75h3.00568C11.6666 0 12 .336341 12 .75123866v3.00567786C12 4.1673102 11.6671 4.5 11.25 4.5c-.41421 0-.75-.3382314-.75-.74308348V2.625L6.53246 6.59254c-.28982.2898217-.76955.2953689-1.06202.0029022l-.06588-.0658844c-.29098-.2909742-.29117-.7679483.0029-1.0620178L9.375 1.5H8.24308zm-6.74753 0C.67089 1.5 0 2.1695784 0 2.99554521v7.50890959C0 11.3291147.66958 12 1.49555 12h7.5089C9.82911 12 10.5 11.3304216 10.5 10.5044548V6.75H9v3.75H1.5V3h3.75V1.5H1.49555z"></path>
  </g>
</svg>
`;

exports[`KIcon renders featureRequest icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-featureRequest">
  <title>featureRequest</title>
  <g>
    <path fill="#BFBFBF" fill-rule="nonzero" d="M7 11l-3 3v-3H2c-1.1045695 0-2-.895431-2-2V2C0 .895431.8954305 0 2 0h12c1.1045695 0 2 .895431 2 2v7c0 1.104569-.8954305 2-2 2H7zM2 2v7h12V2H2z"></path>
  </g>
</svg>
`;

exports[`KIcon renders gateway icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-gateway">
  <title>gateway</title>
  <g>
    <path d="M14 0h-4a2 2 0 0 0-2 2v20a2 2 0 0 0 2 2h4a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2zm0 2v20h-4V2h4zm6 0v2h-3v2h3v2l4-3-4-3zm0 14v2h-3v2h3v2l4-3-4-3zM7 11H4V9l-4 3 4 3v-2h3v-2zm0-7H3v2h4V4zm0 14H3v2h4v-2zm-5 0H0v2h2v-2zm19-7h-4v2h4v-2zm3 0h-2v2h2v-2zM2 4H0v2h2V4z" fill="#A3BBCC" fill-rule="evenodd"></path>
  </g>
</svg>
`;

exports[`KIcon renders gear icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-gear">
  <title>gear</title>
  <g>
    <path fill="#000" fill-opacity=".55" fill-rule="evenodd" d="M11.4765344 3.1092521l1.4732131-1.4732131 1.4142135 1.4142135-1.4732131 1.4732131c.514815.7229326.8733467 1.5648778 1.0262937 2.4765344H16v2h-2.0829584c-.152947.9116566-.5114787 1.7536018-1.0262937 2.4765344l1.4732131 1.4732131-1.4142135 1.4142135-1.4732131-1.4732131c-.7229326.514815-1.5648778.8733467-2.4765344 1.0262937V16H7v-2.0829584c-.9116566-.152947-1.7536018-.5114787-2.4765344-1.0262937L3.0502525 14.363961 1.636039 12.9497475l1.4732131-1.4732131C2.5944371 10.7536018 2.2359054 9.9116566 2.0829584 9H0V7h2.0829584c.152947-.9116566.5114787-1.7536018 1.0262937-2.4765344L1.636039 3.0502525 3.0502525 1.636039l1.4732131 1.4732131C5.2463982 2.5944371 6.0883434 2.2359054 7 2.0829584V0h2v2.0829584c.9116566.152947 1.7536018.5114787 2.4765344 1.0262937zM8 12c2.209139 0 4-1.790861 4-4s-1.790861-4-4-4-4 1.790861-4 4 1.790861 4 4 4z" opacity=".5"></path>
  </g>
</svg>
`;

exports[`KIcon renders info icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-info">
  <title>info</title>
  <g>
    <path fill="#FFF" fill-rule="evenodd" d="M16 8c0-4.418278-3.58172-8-8-8S0 3.581722 0 8s3.58172 8 8 8 8-3.581722 8-8zm-1.5 0c0 3.5898509-2.91015 6.5-6.5 6.5S1.5 11.5898509 1.5 8 4.41015 1.5 8 1.5s6.5 2.9101491 6.5 6.5zM6 12h4v-1H9V7H6v1h1v3H6v1zm1-6.5046844C7 5.7740451 7.21404 6 7.50468 6h.99064C8.77405 6 9 5.785965 9 5.4953156v-.9906312C9 4.2259549 8.78596 4 8.49532 4h-.99064C7.22595 4 7 4.214035 7 4.5046844v.9906312z" opacity=".60000002"></path>
  </g>
</svg>
`;

exports[`KIcon renders portal icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-portal">
  <title>portal</title>
  <g>
    <path d="M2 24h18a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2a2 2 0 0 0-2 2v20a2 2 0 0 0 2 2zm0-2V2h18v20H2zm3-7h12v-2H5v2zm0-4h5V9H5v2zm7 0h5V9h-5v2zM5 7h12V5H5v2zm0 12h5v-2H5v2zm7 0h5v-2h-5v2z" fill="#A3BBCC" fill-rule="evenodd"></path>
  </g>
</svg>
`;

exports[`KIcon renders search icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-search">
  <title>search</title>
  <g>
    <path fill="#B9BABB" fill-rule="evenodd" d="M9.1237341 11.1237341C8.2138377 11.6796483 7.1443299 12 6 12c-3.3137085 0-6-2.6862915-6-6s2.6862915-6 6-6 6 2.6862915 6 6c0 1.1443299-.3203517 2.2138377-.8762659 3.1237341L13.5 11.5c.577204.577204.5829926 1.4170074 0 2-.5781647.5781647-1.4256897.5743103-2 0l-2.3762659-2.3762659zM6 10c2.209139 0 4-1.790861 4-4S8.209139 2 6 2 2 3.790861 2 6s1.790861 4 4 4z"></path>
  </g>
</svg>
`;

exports[`KIcon renders security icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-security">
  <title>security</title>
  <g>
    <path d="M10 24C5.39 22.156.994 19.54.147 14h2.02c.643 3.577 3.15 5.847 7.832 7.837 4.576-1.955 7.15-4.307 7.822-7.837h2.026c-.869 5.45-5.3 8.181-9.847 10zM20 8h-2V5.862c-2.704-.373-5.375-1.501-8-3.359-2.625 1.858-5.296 2.986-8 3.36V8H0V4c3.333 0 6.667-1.333 10-4 3.333 2.667 6.667 4 10 4v4zM0 10h20v2H0v-2z" fill="#A3BBCC" fill-rule="nonzero"></path>
  </g>
</svg>
`;

exports[`KIcon renders spinner icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-spinner">
  <title>spinner</title>
  <g>
    <path fill="#fff" d="M13.3154759 4.60336552c.6394482-.63889655.6394482-1.67503449 0-2.31393104-.6388966-.63889655-1.6744828-.63889655-2.3133793 0-.6394483.63889655-.6394483 1.67503449 0 2.31393104.6388965.63889655 1.6744827.63889655 2.3133793 0" opacity=".1"></path>
    <path fill="#fff" d="M13.9609931 6.17246897c-.9037241 0-1.6364138.73268965-1.6364138 1.63641379 0 .90317241.7326897 1.63586207 1.6364138 1.63586207.9037241 0 1.6358621-.73268966 1.6358621-1.63586207 0-.90372414-.732138-1.63641379-1.6358621-1.63641379" opacity=".25"></path>
    <path fill="#fff" d="M10.9938207 11.0105379c-.6388966.6388966-.6388966 1.6750345 0 2.3139311.6388965.6388965 1.6750345.6388965 2.313931 0 .6388966-.6388966.6388966-1.6750345 0-2.3139311-.6388965-.6388965-1.6750345-.6388965-2.313931 0" opacity=".35"></path>
    <path fill="#fff" d="M7.79988966 12.3247448c-.90372414 0-1.63586207.7326897-1.63586207 1.6364138 0 .9031724.73213793 1.6358621 1.63586207 1.6358621.90372413 0 1.63641379-.7326897 1.63641379-1.6358621 0-.9037241-.73268966-1.6364138-1.63641379-1.6364138" opacity=".5"></path>
    <path fill="#fff" d="M2.28706207 11.004469c-.63889655.6388965-.63889655 1.6750344 0 2.313931.63889655.6388966 1.67503448.6388966 2.31393103 0 .63889656-.6388966.63889656-1.6750345 0-2.313931-.63889655-.6388966-1.67503448-.6388966-2.31393103 0" opacity=".65"></path>
    <path fill="#fff" d="M3.27244138 7.80888276c0-.90372414-.73268966-1.63641379-1.63641379-1.63641379-.90372414 0-1.63586207.73268965-1.63586207 1.63641379 0 .90317241.73213793 1.63586207 1.63586207 1.63586207.90372413 0 1.63641379-.73268966 1.63641379-1.63586207" opacity=".75"></path>
    <path fill="#fff" d="M4.59271724 4.60943448c.63889655-.63889655.63889655-1.67503448 0-2.31393103-.63889655-.63889655-1.67503448-.63889655-2.31393103 0-.63889655.63889655-.63889655 1.67503448 0 2.31393103.63889655.63889655 1.67503448.63889655 2.31393103 0" opacity=".85"></path>
    <path fill="#fff" d="M6.16402759 1.6361931c0 .90372414.73213793 1.6364138 1.63586207 1.6364138.90372413 0 1.63641379-.73268966 1.63641379-1.6364138 0-.90372413-.73268966-1.63641379-1.63641379-1.63641379-.90372414 0-1.63586207.73268966-1.63586207 1.63641379z" opacity=".95"></path>
  </g>
</svg>
`;

exports[`KIcon renders support icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-support">
  <title>support</title>
  <g>
    <path fill="#BFBFBF" fill-rule="evenodd" d="M3 12v1c0 .5522847.4477153 1 1 1h4c.5522847 0 1 .4477153 1 1s-.4477153 1-1 1H4c-1.6568542 0-3-1.3431458-3-3v-1c-.5522847 0-1-.4477153-1-1V7c0-.5522847.4477153-1 1-1h.0708889C1.5561185 2.6077059 4.4735281 0 8 0s6.4438815 2.6077059 6.9291111 6H15c.5522847 0 1 .4477153 1 1v4c0 .5522847-.4477153 1-1 1h-2c-.5522847 0-1-.4477153-1-1V7c0-.5188639.3951689-.9454312.9009623-.9951576C12.439565 3.7201899 10.4206685 2 8 2 5.5793315 2 3.560435 3.7201899 3.0990377 6.0048424 3.6048311 6.0545688 4 6.4811361 4 7v4c0 .5522847-.4477153 1-1 1z"></path>
  </g>
</svg>
`;

exports[`KIcon renders trash icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-trash">
  <title>trash</title>
  <g>
    <path d="M3 3V2c0-1.1045695.8954305-2 2-2h4c1.1045695 0 2 .8954305 2 2v1h2c.5522847 0 1 .4477153 1 1v1H0V4c0-.5522847.4477152-1 1-1h2zm2 0V2h4v1H5zM1 6h2v8h8V6h2v8c0 1.1045695-.8954305 2-2 2H3c-1.1045695 0-2-.8954305-2-2V6zm5 0v7c-.5522847 0-1-.4477153-1-1V7c0-.5522847.4477153-1 1-1zm2 0c.5522847 0 1 .4477153 1 1v5c0 .5522847-.4477153 1-1 1V6z" fill="#BFBFBF" fill-rule="evenodd"></path>
  </g>
</svg>
`;

exports[`KIcon renders vitals icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-vitals">
  <title>vitals</title>
  <g>
    <path d="M19.593 5.493l-2.842 2.842A6.002 6.002 0 0 1 13 17.917l-.001 4.034c5.053-.502 9-4.766 9-9.951a9.96 9.96 0 0 0-2.407-6.507zM18.146 4.11A9.95 9.95 0 0 0 13 2.05v4.034c.83.14 1.604.45 2.281.893l2.865-2.865zM2.05 13A10.003 10.003 0 0 0 11 21.95v-4.033A6.005 6.005 0 0 1 6.083 13H2.049zm0-2h4.034A6.005 6.005 0 0 1 11 6.083V2.049A10.003 10.003 0 0 0 2.05 11zM12 24C5.373 24 0 18.627 0 12S5.373 0 12 0s12 5.373 12 12-5.373 12-12 12zm0-8a4 4 0 1 0 0-8 4 4 0 0 0 0 8z" fill="#A3BBCC" fill-rule="evenodd"></path>
  </g>
</svg>
`;

exports[`KIcon renders workspaces icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-workspaces">
  <title>workspaces</title>
  <g>
    <path fill="#A3BBCC" fill-rule="evenodd" d="M1 0h5c.55228 0 1 .44772 1 1v5c0 .55228-.44772 1-1 1H1c-.55228 0-1-.44772-1-1V1c0-.55228.44772-1 1-1zm1 2v3h3V2H2zM1 9h5c.55228 0 1 .44772 1 1v5c0 .55228-.44772 1-1 1H1c-.55228 0-1-.44772-1-1v-5c0-.55228.44772-1 1-1zm1 2v3h3v-3H2zm8-11h5c.55228 0 1 .44772 1 1v5c0 .55228-.44772 1-1 1h-5c-.55228 0-1-.44772-1-1V1c0-.55228.44772-1 1-1zm1 2v3h3V2h-3zm-1 7h5c.55228 0 1 .44772 1 1v5c0 .55228-.44772 1-1 1h-5c-.55228 0-1-.44772-1-1v-5c0-.55228.44772-1 1-1zm1 2v3h3v-3h-3z"></path>
  </g>
</svg>
`;

exports[`KIcon renders workspacesCollapsed icon 1`] = `
<svg height="24" width="24" viewBox="0 0 24 24" role="img" class="kong-icon kong-icon-workspacesCollapsed">
  <title>workspacesCollapsed</title>
  <g>
    <path d="M2 0h6a2 2 0 0 1 2 2v6a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V2a2 2 0 0 1 2-2zm0 2v6h6V2H2zm0 10h6a2 2 0 0 1 2 2v6a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2v-6a2 2 0 0 1 2-2zm0 2v6h6v-6H2zM14 0h6a2 2 0 0 1 2 2v6a2 2 0 0 1-2 2h-6a2 2 0 0 1-2-2V2a2 2 0 0 1 2-2zm0 2v6h6V2h-6zm1 10l5 5-5 5V12z" fill="#A3BBCC" fill-rule="evenodd"></path>
  </g>
</svg>
`;
